applications/CoSimulationApplication/python_scripts/convergence_accelerators/aitken.py

```diff
@@ -48,33 +50,43 @@ def UpdateSolution( self, r, x ):
         if k == 0:
             alpha = min( self.alpha_old, self.init_alpha_max )
             if self.echo_level > 3:
-                cs_tools.cs_print_info(self._ClassName(), ": Doing relaxation in the first iteration with initial factor = " + "{0:.1g}".format(alpha))
+                cs_tools.cs_print_info(self._ClassName(), ": Doing relaxation in the first iteration with initial factor = {}".format(alpha))
             return alpha * r
         else:
             r_diff = self.R[0] - self.R[1]
             numerator = np.inner( self.R[1], r_diff )
             denominator = np.inner( r_diff, r_diff )
             alpha = -self.alpha_old * numerator/denominator
             if self.echo_level > 3:
-                cs_tools.cs_print_info(self._ClassName(), ": Doing relaxation with factor = " + "{0:.1g}".format(alpha))
-            if alpha > 20:
-                alpha = 20
+                cs_tools.cs_print_info(self._ClassName(), ": Doing relaxation with factor = {}".format(alpha))
+            if alpha > self.alpha_max:
+                alpha = self.alpha_max
                 if self.echo_level > 0:
-                    cs_tools.cs_print_warning(self._ClassName(), "dynamic relaxation factor reaches upper bound: 20")
-            elif alpha < -2:
-                alpha = -2
+                    cs_tools.cs_print_warning(self._ClassName(), "dynamic relaxation factor reaches upper bound: {}".format(self.alpha_max))
+            elif alpha < self.alpha_min:
+                alpha = self.alpha_min
                 if self.echo_level > 0:
-                    cs_tools.cs_print_warning(self._ClassName(), "dynamic relaxation factor reaches lower bound: -2")
+                    cs_tools.cs_print_warning(self._ClassName(), "dynamic relaxation factor reaches lower bound: {}".format(self.alpha_min))
             delta_x = alpha * self.R[0]
         self.alpha_old = alpha
 
         return delta_x
 
+    # Finalizes the current time step and initializes the next time step.
+    def FinalizeSolutionStep( self ):
+        ## Clear the buffer
+        if self.R:
+            if self.echo_level > 3:
+                cs_tools.cs_print_info(self._ClassName(), "Cleaning")
+            self.R.clear()
+
     @classmethod
     def _GetDefaultSettings(cls):
         this_defaults = KM.Parameters("""{
             "init_alpha"     : 0.1,
-            "init_alpha_max" : 0.45
+            "init_alpha_max" : 0.45,
+            "alpha_max"      : 20.0,
```

Comment:
by the way can we decreased the alpha_max to 2 ... 20 is a bit too much I think. No ?

Reply:
agree, @shahrokh-shayegan also said the same

Reply:
done